.github/workflows/sync-fork.yml

```
@@ -0,0 +1,30 @@
+name: Sync Fork with Upstream
+on:
+  schedule:
+    - cron: '0 0 * * *'
+  workflow_dispatch:
+
+jobs:
+  sync:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Forked Repository
+        uses: actions/checkout@v4
+        with:
+            fetch-tags: true
+            token: ${{ secrets.SYNC_FORK_TOKEN }}
+
+      - name: Configure comitter
+        run: |
+          git config --global user.email "deployer@leukeleu.nl"
+          git config --global user.name "Leukeleu bot"
+
+      - name: Add Upstream Remote
+        run: git remote add upstream https://github.com/astral-sh/setup-uv.git
+
+      - name: Fetch Upstream Changes
+        run: git fetch --prune --all --verbose
+
+      - name: Push changes to origin
+        run: |
+          git push origin refs/heads/main --tags --verbose
```

The following should also work, I just ran this locally and pushed all current tags to leukeleu/setup-uv: https://github.com/leukeleu/setup-uv/tags
Some of these are indeed not part of a branch, i.e. not part of the `main` branch in our fork: https://github.com/leukeleu/setup-uv/commit/e92bafb6253dcd438e0484186d7669ea7a8ca1cc

```suggestion
        run: git fetch upstream --prune --tags --verbose

      - name: Push changes to origin
        run: git push origin --tags --verbose
```

So merging this should indeed work, we wont be updating the `main` branch, only the tags.